xml/src/main/scala/akka/stream/alpakka/xml/impl/StreamingXmlParser.scala

```diff
@@ -55,11 +56,13 @@ import scala.annotation.tailrec
       @tailrec private def advanceParser(): Unit =
         if (parser.hasNext) {
           parser.next() match {
-            case AsyncXMLStreamReader.EVENT_INCOMPLETE =>
-              if (!isClosed(in)) pull(in)
-              else failStage(new IllegalStateException("Stream finished before event was fully parsed."))
+            case AsyncXMLStreamReader.EVENT_INCOMPLETE if isClosed(in) && !started => completeStage()
```

Comment:
In general, I like the checks on left hand side of `=>` (actions on the right hand side of `=>`). If it is not good then I'll convert it to `if elseif else`